src/ILLink.Tasks/CreateRuntimeRootDescriptorFile.cs

```diff
@@ -336,5 +396,15 @@ string GetFullClassName (string classNamespace, string className)
 
 			return namespaceDictionary[classNamespace] + "." + className;
 		}
+
+		void ParseDefineConstants ()
+		{
+			if (DefineConstants is not null) {
+				foreach (var item in DefineConstants)
+					defineConstants.Add (item.ItemSpec.Trim ());
+			}
+
+			defineConstants.Add ("FOR_ILLINK");
```

Comment:
Instead of building this in, might it be better to pass it in from runtime?

Reply:
Good point